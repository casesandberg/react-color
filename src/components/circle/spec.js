/* global test, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme';

import Circle from './Circle'
import CircleSwatch from './CircleSwatch'
import { Swatch } from '../common'
import color from '../../helpers/color'

test('Circle renders correctly', () => {
  const tree = renderer.create(
    <Circle />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Circle onChange events correctly', () => {
  const changeSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy()
  })
  const tree = mount(
    <Circle onChange={changeSpy} />
  )
  expect(changeSpy).toHaveBeenCalledTimes(0)
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate('click')

  expect(changeSpy).toHaveBeenCalled()
})


test('Circle with onSwatchHover events correctly', () => {
  const hoverSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy()
  })
  const tree = mount(
    <Circle onSwatchHover={hoverSpy} />
  )
  expect(hoverSpy).toHaveBeenCalledTimes(0)
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate('mouseOver')

  expect(hoverSpy).toHaveBeenCalled()
})

test('CircleSwatch renders correctly', () => {
  const tree = renderer.create(
    <CircleSwatch />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('CircleSwatch renders with sizing and spacing', () => {
  const tree = renderer.create(
    <CircleSwatch circleSize={ 40 } circleSpacing={ 40 } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
