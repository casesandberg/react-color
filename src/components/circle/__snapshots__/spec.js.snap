exports[`test Circle renders correctly 1`] = `
<div
  className="circle-picker"
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
      "marginBottom": -14,
      "marginRight": -14,
      "width": "252px",
    }
  }>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #f44336",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #f44336",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #f44336",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #f44336",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #f44336",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#f44336" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #e91e63",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #e91e63",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #e91e63",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #e91e63",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #e91e63",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#e91e63" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #9c27b0",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #9c27b0",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #9c27b0",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #9c27b0",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #9c27b0",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#9c27b0" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #673ab7",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #673ab7",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #673ab7",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #673ab7",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #673ab7",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#673ab7" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #3f51b5",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #3f51b5",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #3f51b5",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #3f51b5",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #3f51b5",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#3f51b5" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #2196f3",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #2196f3",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #2196f3",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #2196f3",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #2196f3",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#2196f3" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #03a9f4",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #03a9f4",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #03a9f4",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #03a9f4",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #03a9f4",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#03a9f4" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #00bcd4",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #00bcd4",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #00bcd4",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #00bcd4",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #00bcd4",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#00bcd4" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #009688",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #009688",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #009688",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #009688",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #009688",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#009688" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #4caf50",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #4caf50",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #4caf50",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #4caf50",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #4caf50",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#4caf50" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #8bc34a",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #8bc34a",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #8bc34a",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #8bc34a",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #8bc34a",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#8bc34a" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #cddc39",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #cddc39",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #cddc39",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #cddc39",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #cddc39",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#cddc39" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #ffeb3b",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #ffeb3b",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #ffeb3b",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #ffeb3b",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #ffeb3b",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#ffeb3b" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #ffc107",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #ffc107",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #ffc107",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #ffc107",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #ffc107",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#ffc107" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #ff9800",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #ff9800",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #ff9800",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #ff9800",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #ff9800",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#ff9800" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #ff5722",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #ff5722",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #ff5722",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #ff5722",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #ff5722",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#ff5722" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #795548",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #795548",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #795548",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #795548",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #795548",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#795548" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "MozTransform": "scale(1)",
          "MozTransition": "100ms transform ease",
          "OTransform": "scale(1)",
          "OTransition": "100ms transform ease",
          "WebkitTransform": "scale(1)",
          "WebkitTransition": "100ms transform ease",
          "height": 28,
          "marginBottom": 14,
          "marginRight": 14,
          "msTransform": "scale(1)",
          "msTransition": "100ms transform ease",
          "transform": "scale(1)",
          "transition": "100ms transform ease",
          "width": 28,
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "MozBorderRadius": "50%",
            "MozBoxShadow": "inset 0 0 0 14px #607d8b",
            "MozTransition": "100ms box-shadow ease",
            "OBorderRadius": "50%",
            "OBoxShadow": "inset 0 0 0 14px #607d8b",
            "OTransition": "100ms box-shadow ease",
            "WebkitBorderRadius": "50%",
            "WebkitBoxShadow": "inset 0 0 0 14px #607d8b",
            "WebkitTransition": "100ms box-shadow ease",
            "background": "transparent",
            "borderRadius": "50%",
            "boxShadow": "inset 0 0 0 14px #607d8b",
            "cursor": "pointer",
            "height": "100%",
            "msBorderRadius": "50%",
            "msBoxShadow": "inset 0 0 0 14px #607d8b",
            "msTransition": "100ms box-shadow ease",
            "transition": "100ms box-shadow ease",
            "width": "100%",
          }
        }
        title="#607d8b" />
    </div>
  </span>
</div>
`;

exports[`test CircleSwatch renders correctly 1`] = `
<span
  onMouseOut={[Function]}
  onMouseOver={[Function]}>
  <div
    style={
      Object {
        "MozTransform": "scale(1)",
        "MozTransition": "100ms transform ease",
        "OTransform": "scale(1)",
        "OTransition": "100ms transform ease",
        "WebkitTransform": "scale(1)",
        "WebkitTransition": "100ms transform ease",
        "height": 28,
        "marginBottom": 14,
        "marginRight": 14,
        "msTransform": "scale(1)",
        "msTransition": "100ms transform ease",
        "transform": "scale(1)",
        "transition": "100ms transform ease",
        "width": 28,
      }
    }>
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "50%",
          "MozBoxShadow": "inset 0 0 0 14px undefined",
          "MozTransition": "100ms box-shadow ease",
          "OBorderRadius": "50%",
          "OBoxShadow": "inset 0 0 0 14px undefined",
          "OTransition": "100ms box-shadow ease",
          "WebkitBorderRadius": "50%",
          "WebkitBoxShadow": "inset 0 0 0 14px undefined",
          "WebkitTransition": "100ms box-shadow ease",
          "background": "transparent",
          "borderRadius": "50%",
          "boxShadow": "inset 0 0 0 14px undefined",
          "cursor": "pointer",
          "height": "100%",
          "msBorderRadius": "50%",
          "msBoxShadow": "inset 0 0 0 14px undefined",
          "msTransition": "100ms box-shadow ease",
          "transition": "100ms box-shadow ease",
          "width": "100%",
        }
      }
      title={undefined} />
  </div>
</span>
`;

exports[`test CircleSwatch renders with sizing and spacing 1`] = `
<span
  onMouseOut={[Function]}
  onMouseOver={[Function]}>
  <div
    style={
      Object {
        "MozTransform": "scale(1)",
        "MozTransition": "100ms transform ease",
        "OTransform": "scale(1)",
        "OTransition": "100ms transform ease",
        "WebkitTransform": "scale(1)",
        "WebkitTransition": "100ms transform ease",
        "height": 40,
        "marginBottom": 40,
        "marginRight": 40,
        "msTransform": "scale(1)",
        "msTransition": "100ms transform ease",
        "transform": "scale(1)",
        "transition": "100ms transform ease",
        "width": 40,
      }
    }>
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "50%",
          "MozBoxShadow": "inset 0 0 0 20px undefined",
          "MozTransition": "100ms box-shadow ease",
          "OBorderRadius": "50%",
          "OBoxShadow": "inset 0 0 0 20px undefined",
          "OTransition": "100ms box-shadow ease",
          "WebkitBorderRadius": "50%",
          "WebkitBoxShadow": "inset 0 0 0 20px undefined",
          "WebkitTransition": "100ms box-shadow ease",
          "background": "transparent",
          "borderRadius": "50%",
          "boxShadow": "inset 0 0 0 20px undefined",
          "cursor": "pointer",
          "height": "100%",
          "msBorderRadius": "50%",
          "msBoxShadow": "inset 0 0 0 20px undefined",
          "msTransition": "100ms box-shadow ease",
          "transition": "100ms box-shadow ease",
          "width": "100%",
        }
      }
      title={undefined} />
  </div>
</span>
`;
